### PR TITLE
Optimize pack-manager for style&lint

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/asset/asset-manager/pack-manager.ts [hunk 1]
```diff
@@ -25,17 +25,22 @@
...
 import RequestItem from './request-item';
 import { files } from './shared';
 
-export type Unpacker = (packUuid: string[], data: any, options: Record<string, any>, onComplete: ((err: Error | null, data?: any | null) => void)) => void;
+export type Unpacker = (
+    packUuid: string[],
+    data: any,
+    options: Record<string, any>,
+    onComplete: ((err: Error | null, data?: any) => void),
+) => void;
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:40]: Lint problem: too long line

File: cocos/asset/asset-manager/pack-manager.ts [hunk 2]
```diff
@@ -25,17 +25,22 @@
...
 
 interface IUnpackRequest {
-    onComplete: ((err: Error | null, data?: any | null) => void);
+    onComplete: ((err: Error | null, data?: any) => void);
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:43]: Lint problem: `any | null` is equivalent to `any`.

File: cocos/asset/asset-manager/pack-manager.ts [hunk 3]
```diff
@@ -73,12 +78,17 @@ export class PackManager {
      * });
      *
      */
-    public unpackJson (pack: string[], json: any, options: Record<string, any>, onComplete: ((err: Error | null, data?: Record<string, any> | null) => void)): void {
-        let out = js.createMap(true);
+    public unpackJson (
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:81]: Lint problem: too long line

File: cocos/asset/asset-manager/pack-manager.ts [hunk 4]
```diff
@@ -73,12 +78,17 @@ export class PackManager {
...
+        pack: string[],
+        json: any,
+        options: Record<string, any>,
+        onComplete: ((err: Error | null, data?: Record<string, any> | null) => void),
+    ): void {
+        const out: Record<string, any> = js.createMap(true);
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:87]: Problem: untyped thing.

File: cocos/asset/asset-manager/pack-manager.ts [hunk 5]
```diff
@@ -73,12 +78,17 @@ export class PackManager {
...
         let err: Error | null = null;
 
         if (Array.isArray(json)) {
-            json = unpackJSONs(json as any);
+            json = unpackJSONs(json as Parameters<typeof unpackJSONs>[0]);
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts]: Problem: type cast

File: cocos/asset/asset-manager/pack-manager.ts [hunk 6]
```diff
@@ -107,7 +117,8 @@ export class PackManager {
                 }
             } else {
                 err = new Error('unmatched type pack!');
-                out = null;
+                onComplete(err, null);
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:120]: Problem: if don't do this, the `out` should be `null | Record<...>` which then causes many assertions in above.

File: cocos/asset/asset-manager/pack-manager.ts [hunk 7]
```diff
@@ -170,7 +181,13 @@ export class PackManager {
      * });
      *
      */
-    public unpack (pack: string[], data: any, type: string, options: Record<string, any>, onComplete: ((err: Error | null, data?: any | null) => void)): void {
+    public unpack (
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:184]: Lint problem: too long line

File: cocos/asset/asset-manager/pack-manager.ts [hunk 8]
```diff
@@ -200,7 +217,7 @@ export class PackManager {
      * packManager.load(requestItem, null, (err, data) => console.log(err));
      *
      */
-    public load (item: RequestItem, options: Record<string, any> | null, onComplete: ((err: Error | null, data?: any | null) => void)): void {
+    public load (item: RequestItem, options: Record<string, any> | null, onComplete: ((err: Error | null, data?: any) => void)): void {
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:220]: Lint problem: any | null is equivalent to any.

File: cocos/asset/asset-manager/pack-manager.ts [hunk 9]
```diff
@@ -215,35 +232,39 @@ export class PackManager {
         const packs = item.info.packs;
 
         // find a loading package
-        let pack = packs.find((val): boolean => this._loading.has(val.uuid));
+        const loadingPack = packs.find((val): boolean => this._loading.has(val.uuid));
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:235]: To avoid below from `pack!` assertion.

File: cocos/asset/asset-manager/pack-manager.ts [hunk 10]
```diff
@@ -215,35 +232,39 @@ export class PackManager {
...
                 if (!err2) {
                     for (const id in result) {
                         files.add(id, result[id]);
                     }
                 }
-                const callbacks = this._loading.remove(pack!.uuid);
-                for (let i = 0, l = callbacks!.length; i < l; i++) {
-                    const cb = callbacks![i];
+                const callbacks = this._loading.remove(pack.uuid);
+                assertIsTrue(callbacks);
```
Comment (shrinktofit) [inline cocos/asset/asset-manager/pack-manager.ts:265]: Prefer explicit assertion.

**Review by SantyWang — APPROVED** [review]

**Pull request merged**